Sources/ProjectSpec/Linkage.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+public enum Linkage {
+    case dynamic
+    case `static`
+    case none
+}
+
+extension Target {
+    
+    public var defaultLinkage: Linkage {
+        switch type {
+        case .none,
+             .appExtension,
+             .application,
+             .bundle,
+             .commandLineTool,
+             .messagesApplication,
+             .messagesExtension,
+             .ocUnitTestBundle,
+             .stickerPack,
+             .tvExtension,
+             .uiTestBundle,
+             .unitTestBundle,
+             .watchApp,
+             .watchExtension,
+             .watch2App,
+             .watch2Extension,
+             .xcodeExtension,
+             .xpcService:
+            return .none
+        case .framework:
+            // TODO: This should check `MACH_O_TYPE` in case this is a "Static Framework"
```

Comment:
I'd like to get this in. This logic should then probably be moved into XcodeGenKit

Reply:
The only reason I didn’t is because just like #350 is all about static libraries there is another class of changes needed for static frameworks. I didn’t want to put part of the static frameworks change in with this, especially since it wasn’t the purpose. The reason for a TODO is because I plan on doing that very soon (as soon as #350 is done, so this week).

Reply:
I think the type should stay in ProjectSpec since you will be able to specify it as a `preferredLinkage` (as described in #343). I can move the extension if desired though?

Reply:
Yeah the type can stay here, though the logic in the switch could go into XcodeGen. It would have to move later anyway if you check for `MACH_O_TYPE` within the switch.